database/src/main/java/com/linkall/cdk/database/debezium/DebeziumSource.java

```diff
@@ -14,167 +14,163 @@
-        final Properties debeziumProperties = new Properties();
+    abstract protected CloudEventData convertData(Object data) throws IOException;
 
-        debezium.stream().forEach(k -> {
-            debeziumProperties.put(k.getKey(), debezium.getString(k.getKey()));
-        });
-
-        return debeziumProperties;
-    }
-
-    public Set<String> getExcludedTables(Set<String> excludeTables) {
-        Set<String> exclude = new HashSet<>(getSystemExcludedTables());
-        exclude.addAll(excludeTables);
-        return exclude;
+    @Override
+    final public void destroy() throws Exception {
+        if (engine != null)
+            engine.close();
+        executor.shutdown();
     }
 
-    public String tableFormat(String name, Stream<String> table) {
-        return table
-                .map(stream -> name + "." + stream)
-                .collect(Collectors.joining(","));
+    @Override
+    final public BlockingQueue<Tuple> queue() {
+        return events;
     }
 
     @Override
-    public Class<? extends Config> configClass() {
-        return null;
+    final public void initialize(Config config) throws Exception {
+        Class<? extends Config> c = this.configClass();
+        if (!c.isInstance(config)) {
+            throw new Exception("invalid config class, " + c.getName() + " is needed");
+        }
+        this.debeziumConfig = (DebeziumConfig) config;
+        this.start();
     }
 
     @Override
-    public void initialize(Config config) throws Exception {
-        start();
+    final public void handleBatch(List<ChangeEvent<String, String>> records,
+                            DebeziumEngine.RecordCommitter<ChangeEvent<String, String>> committer)
+            throws InterruptedException {
+        CountDownLatch latch = new CountDownLatch(records.size());
+        for (ChangeEvent<String, String> record : records) {
+            LOGGER.info("Received event '{}'", record);
+            if (record.value() == null) {
+                latch.countDown();
+                continue;
+            }
+            try {
+                CloudEvent ceEvent = this.convert(record.value());
+                events.put(
+                    new Tuple(ceEvent,
+                        () -> commit(latch, record, committer),
+                        (msg) -> {
+                            LOGGER.error("event send failed:{},{}", msg, EventUtil.eventToJson(ceEvent));
+                            commit(latch, record, committer);
+                        }
+                    )
+                );
+            } catch (IOException e) {
+                latch.countDown(); // How to process offset?
+                LOGGER.error("failed to parse record data {} to json, error: {}", record.value(), e);
+            }
+        }
+        latch.await();
+        committer.markBatchFinished();
     }
 
+    final protected void start() {
+        engine = DebeziumEngine.create(CloudEvents.class)
```

Comment:
Debezium original change data is SourceRecord, if using Connect.class it no needs to convert and  serialize the change event as SourceRecord, otherwise, it needs to  convert and  serialize to string, such as Json, CloudEvents.